requirements.txt

```diff
@@ -1,3 +1,9 @@
+httpx>=0.25.0
```

Comment:
This is an implied library shared between `apache-airflow-providers-slack` and some other library in use in the image. It's pinned here to force the `pip` solver to have a route to land on `apache-airflow` version `2.10.3`.

Reply:
this is interesting and unexpected, and I'm curious how we could find/debug if we run into a similar problem in the future. I wouldn't know where to start - do you have a brief version of how you made it to this solution?

Reply:
TLDR: I broke the build process up into pieces, did parts of it manually, observed where it went sideways and gave it hints on what I wanted.

👇 Opinion time --

Very generally, we tend to be aggressive in how we pin versions in our `requirements.txt`, `packages.json`, etc. It's a total double edged blade -- you get assurance that you won't get tagged by a breaking change, but it becomes easy to fall behind and slip out of that zone where you're running the same stuff as everyone else which tends to work together because they are temporally correlated. 

👇 Fact time --

Influenced by this opinion, I suspected it was something we had in our config, and to confirm that, I spun up our airflow image including overriding the `entrypoint` of the image so that it simply dropped me into a shell without anything else happening. I was able to observe that installed version of `apache-airflow` was `9.3.0`. 

So I built my own image of plain airflow without any of extensions, and it had the right version in it, so I then knew that it was us, which I had suspected re the opinion above. So, I disabled any of our python related extensions to the Dockerfile, spun up a "airflow + our extensions - python stuff" image and hand installed the packages, again working form a shell where I had overloaded the entrypoint. 

It was through this that I was able to directly observe the `pip` solver going south on me, and then it was a matter of installing our packages one a time to see where it got hung up. Turns out that we have two libraries that we install that themselves need `httpx` and based on how they are pinned (or not) in their respective `requirements` files, I determined that it was the solver that wasn't doing what I expected. With that, I pinned this `httpx` where I wanted it, and then it solved "correctly."

I'm not sure if this is a useful description of a general solution for python library wackiness, but it is the solution for this particular issue, at least for me.

Reply:
Thanks for working through this, Frank. I'm not sure if I'm fully following the issues we're running into here, but fwiw I tried clearing the `requirements.txt` of all dependencies except `onepasswordconnectsdk`, and the image built normally.

When i run `docker run -it --rm apache/airflow:2.10.3 bash` and then use `pip freeze` to inspect the python env, I see:

```
apache-airflow==2.10.3
apache-airflow-providers-slack==8.9.1
docker==7.1.0
httpx==0.23.3
```

Sorry if this is already a known dead-end, but it looks to me like the Airflow base image may already have this problem solved 🤷

Reply:
Interesting - thanks for testing this out. I'm seeing something similar. I took it on faith that since we had those extra libraries in our own `requirements.txt` that we used to extend the airflow image, we would continue to need them. I'm paring down our additional requirements, and we can let airflow manage this. Thanks again. https://github.com/cityofaustin/atd-airflow/pull/260/commits/1ebfd61b173cda584101a7ff8a0166f8a6433d98

Reply:
Same—maybe Airflow changed the default python packages that are bundled in the image. I doubt we would have set up that requirements file without good reason to do so.